extensions/spring-scheduling/deployment/src/main/java/io/quarkus/spring/scheduling/deployment/SpringSchedulingProcessor.java

```diff
@@ -0,0 +1,97 @@
+    private static final Logger LOGGER = Logger.getLogger(SpringSchedulingProcessor.class);
+
+    @BuildStep
+    FeatureBuildItem registerFeature() {
+        return new FeatureBuildItem(FeatureBuildItem.SPRING_SCHEDULING);
+    }
+
+    @BuildStep
+    AnnotationsTransformerBuildItem buildScheduledAnnotationTransformer() {
+        return new AnnotationsTransformerBuildItem(new AnnotationsTransformer() {
+            @Override
+            public boolean appliesTo(AnnotationTarget.Kind kind) {
+                return kind == AnnotationTarget.Kind.METHOD;
+            }
+
+            @Override
+            public void transform(TransformationContext context) {
+                MethodInfo methodInfo = context.getTarget().asMethod();
+                List<AnnotationInstance> collect = methodInfo.annotations().stream()
+                        .filter(annotationInstance -> annotationInstance.name().equals(SPRING_SCHEDULED))
+                        .collect(Collectors.toList());
+                if (!collect.isEmpty()) {
+                    context.transform()
+                            .add(QUARKUS_SCHEDULED, AnnotationValue.createStringValue("cron", "0/1 * * * * ?")).done();
+                }
+            }
+        });
+    }
+
+    @BuildStep
+    public void collectSpringScheduledMethods(CombinedIndexBuildItem combinedIndex,
+            BuildProducer<SpringScheduledAnnotatedMethodBuildItem> scheduledBusinessMethods) {
+
+        Map<MethodInfo, AnnotationInstance> result = new HashMap<>();
+
+        // first first go through the list of annotated methods
+        for (AnnotationInstance instance : combinedIndex.getIndex().getAnnotations(SPRING_SCHEDULED)) {
+            if (instance.target().kind() != AnnotationTarget.Kind.METHOD) {
+                throw new IllegalArgumentException("Unsupported annotation target kind " + instance.target().kind().name());
+            }
+            MethodInfo methodInfo = instance.target().asMethod();
+            result.put(methodInfo, instance);
+            LOGGER.debugf("Found Spring scheduled business method %s declared on %s", methodInfo.name(),
+                    methodInfo.declaringClass().name());
+        }
+        scheduledBusinessMethods.produce(new SpringScheduledAnnotatedMethodBuildItem(result));
+    }
+
+    @BuildStep
+    void registerBeans(SpringScheduledAnnotatedMethodBuildItem methodBuildItem,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+        Map<MethodInfo, AnnotationInstance> methodToInstanceMap = methodBuildItem.getMethodToInstanceMap();
+        Set<String> beanClasses = new HashSet<>();
+        for (MethodInfo methodInfo : methodToInstanceMap.keySet()) {
+            beanClasses.add(methodInfo.declaringClass().asClass().name().toString());
+        }
+        additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClasses(beanClasses).setUnremovable()
```

Comment:
This is a little weird TBH - Does regular Spring make a class a bean by simply annotating with `@Scheduled` or does it already have to be a bean (by having `@Component` for example)?

Reply:
Ok. No, Spring does not make a class a bean by annotating with @Scheduled. It has to be already a bean in order to Spring can detect the method annotated with @Scheduled.